dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -276,13 +277,36 @@ DeltaMergeStore::DeltaMergeStore(Context & db_context,
         else
         {
             auto segment_id = DELTA_MERGE_FIRST_SEGMENT_ID;
-            while (segment_id)
+
+            // parallel restore segment to speed up
+            if (thread_pool)
             {
-                auto segment = Segment::restoreSegment(log, *dm_context, segment_id);
-                segments.emplace(segment->getRowKeyRange().getEnd(), segment);
-                id_to_segment.emplace(segment_id, segment);
+                auto wait_group = thread_pool->waitGroup();
+                auto segment_ids = Segment::getAllSegmentIds(*dm_context, segment_id);
+                std::mutex mutex;
```

Comment:
better give it a more specified name, for example `mtx_parallel_init_segments`

Reply:
Or maybe we can reuse the `DeltaMergeStore::read_write_mutex`?

Reply:
Yes, we can reuse read_write_mutex.